### DISCUSSION TITLE
How to flash uart-bridge

### DISCUSSION THREAD

**Discussion by ok5max** [cesanta/mongoose#1914] [Ask questions, get answers] [answered]
Hello,

is there any easy way to flash uart-bridge to ESP32 please?
I don't have linux and mongoose os didn't build it.
Just need uart bridge...

Could you please help me?
Thank you

**Comment by scaprile** [comment, marked as answer]
Please read the README https://github.com/cesanta/mongoose/tree/master/examples/esp32/uart-bridge/#flash-pre-built-firmware , there is a pre-built image for you to flash and links to the flasher for several OSs.

The Makefile is a guide showing what you have to do to build the example, and the Docker image helps us build and test and many users can also use them instead of installing vendor SDK and toolchains.
If you can't take advantage of this, you have to do that yourself. The example is a standard ESP-IDF project https://mongoose.ws/tutorials/esp32/uart-bridge/
Your options are:
- Build a Linux virtual machine and run the example as is
- Install esputil and use the pre-built image
- Install the ESP-IDF and toolchain
  - Install make and do whatever necessary changes for your OS. The [tools setup tutorial](https://mongoose.ws/tutorials/tools/) provides information on installing make
  - Understand the Makefile and build yourself using the ESP-IDF build system; we just call idf.py
